clusterloader2/pkg/measurement/common/metrics_server_measurement.go

```diff
@@ -0,0 +1,345 @@
+	if err != nil {
+		return nil, err
+	}
+	if err = m.stop(); err != nil {
+		return nil, err
+	}
+	// We want to subtract the latest histogram from the first one we collect.
+	finalLatency := HistogramSub(latestLatency, m.initialLatency)
+	// Pretty Print the report.
+	result := &measurementutil.LatencyMetric{}
+	if err = SetQuantileFromHistogram(result, finalLatency); err != nil {
+		return nil, err
+	}
+	content, err := util.PrettyPrintJSON(result)
+	if err != nil {
+		return nil, err
+	}
+	// Create Summary.
+	return []measurement.Summary{measurement.CreateSummary(metricsServerLatencyName, "json", content)}, nil
+}
+
+// startQuerying queries /apis/metrics.k8s.io/v1beta1/pods endpoint every interval
+// and stops when stop() is called.
+func (m *metricsServerLatencyMeasurement) startQuerying(ctx context.Context, client clientset.Interface, interval time.Duration) {
+	m.isRunning = true
+	m.wg.Add(1)
+	go m.queryLoop(ctx, client, interval)
+}
+
+func (m *metricsServerLatencyMeasurement) queryLoop(ctx context.Context, client clientset.Interface, interval time.Duration) {
+	defer m.wg.Done()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(interval):
+			var output string
+			output, err := m.topPods(ctx, client)
+			if err != nil {
+				klog.V(2).Infof("error during fetching pod resource usage: %v", err)
+			}
+			if output == "" {
+				klog.V(2).Infof("/apis/metrics.k8s.io/v1beta1/pods endpoint returns no data")
+			}
+
+		}
+	}
+}
+
+func (m *metricsServerLatencyMeasurement) retrieveMetricsServerLatencyMetrics(kubeConf *rest.Config, metricsServerPodName string) (*measurementutil.Histogram, error) {
+	metricsServerHist := measurementutil.NewHistogram(nil)
+	output, err := m.getMetricsFromMS(kubeConf, metricsServerPodName)
+	if err != nil {
+		return metricsServerHist, err
+	}
+	samples, err := measurementutil.ExtractMetricSamples(output)
+	if err != nil {
+		return metricsServerHist, err
+	}
+	for _, sample := range samples {
+		if isMetricsServerLatencySample(sample) {
+			measurementutil.ConvertSampleToHistogram(sample, metricsServerHist)
+		}
+	}
+	return metricsServerHist, nil
+}
+
+func isMetricsServerLatencySample(sample *model.Sample) bool {
+	name, nameOk := sample.Metric[model.MetricNameLabel]
+	group, groupOK := sample.Metric["group"]
+	resource, resourceOk := sample.Metric["resource"]
+	scope, scopeOk := sample.Metric["scope"]
+
+	if nameOk &&
+		groupOK &&
+		resourceOk &&
+		scopeOk &&
+		name == metricsServerRequestDurationMetricName &&
+		group == metricsServerRequestDurationGroup &&
+		resource == metricsServerRequestDurationResource &&
+		scope == metricsServerScope {
+		return true
+	}
+
+	return false
+}
+
+func (m *metricsServerLatencyMeasurement) getMetricsFromMS(kubeConf *rest.Config, metricsServerPodName string) (string, error) {
+	out, err := proxyRequestToPod(kubeConf, m.namespace, metricsServerPodName, "https", 4443, "/metrics")
```

Comment:
This code assumes that metrics-server exposes port 4443. Unfortunatelly this differs from distribution to distribution and from release to release. 

For scalability tests I would expect that it uses https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/metrics-server configuraiton. This configuration uses port 443. 

 For this PR I would propose to use the port that is used by scalability tests, but we should prepare to be distribution independent by adding code to detect port based on containerPort configuration.